### PR TITLE
NMS-214

### PR DESCRIPTION
**Pull request by srivastavaAbhi**
Enum deserialize fixed.

### CODE REVIEW

File: testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java [hunk 1]
```diff
@@ -109,8 +108,13 @@ private String createFailureResponseJson(String failureReason) throws IOExceptio
     @Test
     public void testCreateFileProcessedStatusRequest() throws IOException, InterruptedException {
         getLogger().debug("testCreateFileProcessedStatusRequest()");
-        HttpPost httpPost = createFileProcessedStatusHttpPost("file.csv",
-                FileProcessedStatus.FILE_PROCESSED_SUCCESSFULLY);
+        String requestJson = "{\"fileProcessedStatus\":8000,\"fileName\":\"file.csv\"}";
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java]: Why not change `createFileProcessedStatusHttpPost()`?

File: testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java [hunk 2]
```diff
@@ -130,8 +134,13 @@ public void testCreateFileProcessedStatusRequestNoStatusCode() throws IOExceptio
     @Test
     public void testCreateFileProcessedStatusRequestNoFileName() throws IOException, InterruptedException {
         getLogger().debug("testCreateFileProcessedStatusRequestNoFileName()");
-        HttpPost httpPost = createFileProcessedStatusHttpPost(null,
-                FileProcessedStatus.FILE_PROCESSED_SUCCESSFULLY);
+
+        String requestJson = "{\"fileProcessedStatus\":8000}";
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java]: Why not change `createFileProcessedStatusHttpPost()`?

File: testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java [hunk 3]
```diff
@@ -142,8 +151,12 @@ public void testCreateFileProcessedStatusRequestNoFileName() throws IOException,
     @Test
     public void testCreateFileProcessedStatusRequestWithError() throws IOException, InterruptedException {
         getLogger().debug("testCreateFileProcessedStatusRequestWithError()");
-        HttpPost httpPost = createFileProcessedStatusHttpPost("file.csv",
-                FileProcessedStatus.FILE_ERROR_IN_FILE_FORMAT);
+        String requestJson = "{\"fileProcessedStatus\":8005,\"fileName\":\"file.csv\"}";
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java]: Why not change `createFileProcessedStatusHttpPost()`?

File: testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java [hunk 4]
```diff
@@ -189,8 +225,13 @@ public void verifyFT200() throws IOException, InterruptedException {
     @Test
     public void verifyFT193() throws IOException, InterruptedException {
         getLogger().debug("testCreateFileProcessedStatusRequestWithErrorFILE_NOT_ACCESSIBLE()");
-        HttpPost httpPost = createFileProcessedStatusHttpPost("file.csv",
-                FileProcessedStatus.FILE_NOT_ACCESSIBLE);
+
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java]: Why not change `createFileProcessedStatusHttpPost()`?

File: testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java [hunk 5]
```diff
@@ -210,8 +251,12 @@ public void verifyFT193() throws IOException, InterruptedException {
     @Test
     public void verifyFT194() throws IOException, InterruptedException {
         getLogger().debug("testCreateFileProcessedStatusRequestWithErrorFILE_CHECKSUM_ERROR()");
-        HttpPost httpPost = createFileProcessedStatusHttpPost("file.csv",
-                FileProcessedStatus.FILE_CHECKSUM_ERROR);
+        String requestJson = "{\"fileProcessedStatus\":8002,\"fileName\":\"file.csv\"}";
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java]: Why not change `createFileProcessedStatusHttpPost()`?

File: testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java [hunk 6]
```diff
@@ -232,8 +277,12 @@ public void verifyFT194() throws IOException, InterruptedException {
     @Test
     public void verifyFT195() throws IOException, InterruptedException {
         getLogger().debug("testCreateFileProcessedStatusRequestWithErrorFILE_RECORDSCOUNT_ERROR()");
-        HttpPost httpPost = createFileProcessedStatusHttpPost("file.csv",
-                FileProcessedStatus.FILE_RECORDSCOUNT_ERROR);
+        String requestJson = "{\"fileProcessedStatus\":8003,\"fileName\":\"file.csv\"}";
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiControllerObdBundleIT.java]: Why not change `createFileProcessedStatusHttpPost()`?

**Comment by frankhuster** [comment]
Thanks for the fix!
Confused why you would not change `createFileProcessedStatusHttpPost()`

**Comment by srivastavaAbhi** [comment]
Will fix it.

**Comment by srivastavaAbhi** [comment]
Now I remember the reason which is weird.
I thought I should have both the flavour, direct send json string and
through serialized as well.
I will make a new method for that repeated work.
Is that fine with you?

**Comment by frankhuster** [comment]
Makes sense if the existing method is still used.

**Comment by srivastavaAbhi** [comment]
It's used but just in one case.

**Comment by frankhuster** [comment]
Cool, then your solution of creating a new method makes sense!

**Pull request merged**